src/control/lib/netdetect/netdetect.go

```diff
@@ -1224,3 +1231,55 @@ func ScanFabric(provider string, excludes ...string) ([]FabricScan, error) {
 	}
 	return ScanResults, nil
 }
+
+// GetDeviceClass determines the device type according to what's stored in the filesystem
+// Returns an integer value corresponding to its ARP protocol hardware identifier
+// found here: https://elixir.free-electrons.com/linux/v4.0/source/include/uapi/linux/if_arp.h#L29
+func GetDeviceClass(netdev string) (uint32, error) {
+	devClass, err := ioutil.ReadFile(fmt.Sprintf("/sys/class/net/%s/type", netdev))
+	if err != nil {
+		return 0, err
+	}
+
+	res, err := strconv.Atoi(strings.TrimSpace(string(devClass)))
+	return uint32(res), err
+}
+
+// Convert a network device class ID to a string identifier according to the table
+// found here: https://elixir.free-electrons.com/linux/v4.0/source/include/uapi/linux/if_arp.h#L29
+func DevClassName(class uint32) string {
+	switch class {
+	case 0:
```

Comment:
I still think these values should be defined in a series of consts in this package.

Reply:
Done.